lazer/evm/test/ExampleReceiver.t.sol

```diff
@@ -12,19 +12,24 @@ contract ExampleReceiverTest is Test {
         address trustedSigner = 0xEfEf56cD66896f6799A90A4e4d512C330c094e44;
         console.log("trustedSigner %s", trustedSigner);
 
+        address lazer = makeAddr("lazer");
         PythLazer pythLazer = new PythLazer();
-        pythLazer.initialize(address(0x1));
-        vm.prank(address(0x1));
+        pythLazer.initialize(lazer);
+        vm.prank(lazer);
         pythLazer.updateTrustedSigner(trustedSigner, 3000000000000000);
 
+        address consumer = makeAddr("consumer");
         ExampleReceiver receiver = new ExampleReceiver(address(pythLazer));
-
         bytes memory update =
             hex"2a22999a577d3cc0202197939d736bc0dcf71b9dde7b9470e4d16fa8e2120c0787a1c0d744d0c39cc372af4d1ecf2d09e84160ca905f3f597d20e2eec144a446a0459ad600001c93c7d3750006240af373971c01010000000201000000000005f5e100";
         console.logBytes(update);
 
-        receiver.updatePrice(update);
+        vm.deal(consumer, 1 ether);
+        vm.prank(consumer);
+        receiver.updatePrice{value: pythLazer.verification_fee()}(update);
```

Comment:
If I attach more fee, the refund fails and I get the following error:
<img width="481" alt="image" src="https://github.com/user-attachments/assets/0e4d24e9-b340-44b8-a4b9-1dcd6870e714" />

If I attach more fee and then comment out the refund in `updatePrice`, the final assert fails saying that consumer has the full amount of ether. Yet, pythLazer was still paid, and now the receiver contract got paid the 9 wei difference. So, money was made out of thin air. 

I honestly don't really understand what I should be doing.

Reply:
@Riateche helped me out. The issue was that when I called vm.prank() before calling the updatePrice function, I was calling pythLazer.verification_fee() to get the fee to attach. This consumed the prank and led to some other default address making the call for verifyUpdate(). This default address doesn't have the receive/fallback functions so it fails. But storing the fee as a uint256 and making sure prank correctly targets the actual call, the test succeeds.